nifi-nar-bundles/nifi-enrich-bundle/nifi-enrich-processors/src/main/java/org/apache/nifi/processors/GeoEnrichIP.java

```diff
@@ -130,6 +132,28 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             getLogger().warn("Failure while trying to load enrichment data for {} due to {}, rolling back session "
                     + "and will reload the database on the next run", flowFile, idbe.getMessage());
             session.rollback();
+            return;
+        } catch (AddressNotFoundException anfe) {
+            session.transfer(flowFile, REL_NOT_FOUND);
+
+            switch (logLevel) {
+                case info:
+                    getLogger().info("Address not found in the database", anfe);
+                    break;
+                case warn:
+                    getLogger().warn("Address not found in the database", anfe);
+                    break;
+                case error:
+                    getLogger().error("Address not found in the database", anfe);
+                    break;
+                case debug:
+                default:
+                    if(getLogger().isDebugEnabled()) {
+                        getLogger().debug("Address not found in the database", anfe);
+                    }
```

Comment:
The `isDebugEnabled()` conditional does not seem necessary since the message does not require any formatting, and the standard logging handlers should filter this operation.
```suggestion
                    getLogger().debug("Address not found in the database", anfe);
```